dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/integration/single/listener/ExportedServiceListener.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.integration.single.listener;
+
+import org.apache.dubbo.config.ServiceConfig;
+import org.apache.dubbo.config.ServiceListener;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * This implementation of {@link ServiceListener} is to record exported services
+ */
+public class ExportedServiceListener implements ServiceListener {
+
+    private List<ServiceConfig> exportedServices = new ArrayList<>(2);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void exported(ServiceConfig sc) {
```

Comment:
internal MetadataService need be handle.

Reply:
We need to check if `MetadataService` exported successfully. Also, I have added more testcases to check `MetadataService`, and use SPI to load `ExportedServiceListener`